src/main/kotlin/com/github/hanseter/json/editor/SchemaNormalizer.kt

```diff
@@ -303,20 +306,29 @@ object SchemaNormalizer {
         if (inlineInItems(subPart, copyTarget)) return
 
         val allOf = subPart.optJSONArray("allOf") ?: return
+
         copyTarget().apply {
             remove("allOf")
-            put("type", "object")
-            put("properties", JSONObject())
         }
-        val order = JSONArray()
-        getAllObjectsInAllOf(allOf).forEach { propObj ->
-            merge(copyTarget().getJSONObject("properties"), propObj.getJSONObject("properties"))
-            propObj.optJSONArray("order")?.also { order.put(it) }
-            propObj.optJSONObject("order")?.also { order.put(it) }
-            merge(copyTarget(), propObj, copyTarget().keySet())
+
+        val order = lazy { JSONArray() }
```

Comment:
Why not use a nullable `JSONArray` here?

Reply:
I thought the assignments in the loop of
```kotlin
if (order == null) order = JSONArray()
order!!.put(it)
```
or
```kotlin
(order ?: JSONArray()).put(it).also { order = it }
```
are both more cumbersome than the `lazy` approach.

Or do you have another idea that hasn't occurred to me?

Reply:
Hm well fair enough.